### PR TITLE
feat(Core/Spell): AssertSpellInfo

### PR DESCRIPTION
**Pull request by Kitzunu**
(cherry picked from commit https://github.com/TrinityCore/TrinityCore/commit/e52878b6b5600c2ed6fe8a60fbc27d220f807a2e)
(cherry picked from commit https://github.com/TrinityCore/TrinityCore/commit/8c44259fae2980598980a7935a3f3941130a5a10)
Co-Authored-By: Shauren <shauren.trinity@gmail.com>
Co-Authored-By: Giacomo Pozzoni <giacomopoz@gmail.com>

needed for https://github.com/azerothcore/azerothcore-wotlk/pull/6109

## Changes Proposed:
-  Use SpellMgr::AssertSpellInfo() if the spell id is valid and always supposed to return a valid not-NULL SpellInfo* 

## Tests Performed:

- none 

## How to Test the Changes:

Test that spells work as they should, you can check Blood Queen encounter that she use SPELL_PACT_OF_THE_DARKFALLEN_DAMAGE

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Spells/SpellMgr.h [hunk 1]
```diff
@@ -684,6 +684,14 @@ class SpellMgr
 
     // SpellInfo object management
     [[nodiscard]] SpellInfo const* GetSpellInfo(uint32 spellId) const { return spellId < GetSpellInfoStoreSize() ?  mSpellInfoMap[spellId] : nullptr; }
+    // Use this only with 100% valid spellIds
+    (uint32 spellId) const
```
Comment (Winfidonarleyan) [inline src/server/game/Spells/SpellMgr.h:688]: ?

Comment (Kitzunu) [reply src/server/game/Spells/SpellMgr.h:688]: Lol

Comment (Kitzunu) [reply src/server/game/Spells/SpellMgr.h]: fixed

**Review by Winfidonarleyan — APPROVED** [review]

**Comment by Kitzunu** [comment]
@azerothcore/testers

**Comment by Sedatyf** [comment]
It's in my backlog @Kitzunu! I'll try to test it as soon as possible

**Comment by tkn963** [comment]
Tested demonic circle many times. While being out of range, it says you can't do that yet. I don't remember how it was back then but shouldn't it say something about being out of range? That's the only thing I found that seemed out of place but most likely not related to this PR
Tested many mounts for skill 75, 150, 225 and 300. Unsure what to specifically look for but I ran around all over the place, switching mounts from time to time and so on
I can't seem to test the Blood-Queens Pact of the Darkfallen because I either have not enough players in the group (3 for 10-man) or I can't see it happening in the combat log or I'm simply doing something wrong

**Comment by Kitzunu** [comment]
The demo circle thing is not related. What would happen if it wouldn't work is that the core would crash =)

**Comment by tkn963** [comment]
> 
> 
> The demo circle thing is not related. What would happen if it wouldn't work is that the core would crash =)
> 
> It would crash if it was not the correct spellId or rather, using a spellid that did not exist

You said to test demonic circle on discord. :p Either way I had no issues with error messages in the console or log and no crashes what so ever

**Comment by Kitzunu** [comment]
Yes, I meant the message you got from it x)

**Pull request merged**